esp32/spiram.h: No such file or directory

Hello, 

I'm trying to build your project with arduino IDE as I'd like to use it on a ESP32 C6 which is not yet supported by platformio.
When building I'm getting the following error : 

```
In file included from C:\Users\Rodolphe\Documents\Arduino\libraries\pzem-edl\src\timeseries.cpp:13:
..\pzem-edl\src\timeseries.hpp:33:10: fatal error: esp32/spiram.h: No such file or directory
   33 | #include <esp32/spiram.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
exit status 1
```
It seems to me that esp32/spiram.h may have moved (or been removed) in the [last version of the framework](https://docs.espressif.com/projects/esp-idf/en/v5.0/esp32s2/migration-guides/release-5.x/system.html#psram)

Hi and thanks for the quick update. 
Copying the last commit files in my src folder, I do not see this error anymore, but I'm getting some new ones which seems quite strange as these are syntax errors : 

`In file included from C:\ESP-32\WattMeter\resources\pzem-edl\examples\pzem_cli\src\timeseries.cpp:13:
timeseries.hpp:281:20: error: expected unqualified-id before ')' token
  281 |     TSContainer<T>(){};
      |                    ^
In file included from C:\ESP-32\WattMeter\resources\pzem-edl\examples\pzem_cli\src\LList.h:22,
                 from C:\ESP-32\WattMeter\resources\pzem-edl\examples\pzem_cli\src\pzem_edl.hpp:18,
                 from C:\ESP-32\WattMeter\resources\pzem-edl\examples\pzem_cli\src\pzem_edl.cpp:13:
LinkedList.h:23:32: error: expected ')' before '*' token
   23 |         ListNode<T>(ListNode<T>* _next = nullptr) : next(_next) {}
      |                    ~           ^
      |                                )
LinkedList.h:24:21: error: expected unqualified-id before 'const'
   24 |         ListNode<T>(const T& _v, ListNode<T>* _next = nullptr) : data(_v), next(_next) {}
      |                     ^~~~~
LinkedList.h:24:21: error: expected ')' before 'const'
   24 |         ListNode<T>(const T& _v, ListNode<T>* _next = nullptr) : data(_v), next(_next) {}
      |                    ~^~~~~
      |                     )
exit status 1
`
I might do something wrong.
Any thoughts about it ?

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -1,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "github-actions" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"

--- a/.github/workflows/pio_build.yml
+++ b/.github/workflows/pio_build.yml
@@ -42,9 +42,9 @@ jobs:
           - "examples/pzem_cli"
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: PIO Cache
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: |
             ~/.cache/pip

--- a/library.json
+++ b/library.json
@@ -19,12 +19,6 @@
         "type": "git",
         "url": "https://github.com/vortigont/pzem-edl.git"
     },
-    "dependencies":
-    [
-         {"owner": "vortigont",
-          "name": "LinkedList",
-          "version": "https://github.com/vortigont/LinkedList"}
-    ],
     "build": {
         "flags": "-std=gnu++14",
         "unflags": "-std=gnu++11"

--- a/src/msgq.cpp
+++ b/src/msgq.cpp
@@ -99,7 +99,7 @@ void UartQ::stop_tx_msg_q(){
 
     // очищаем все сообщения из очереди
     TX_msg* msg = nullptr;
-    while (xQueueReceive(_t, &(msg), (portTickType)0) == pdPASS ){
+    while (xQueueReceive(_t, &(msg), (TickType_t)0) == pdPASS ){
         delete msg;
     }
 

--- a/src/msgq.hpp
+++ b/src/msgq.hpp
@@ -345,7 +345,7 @@ private:
             xSemaphoreGive(rts_sem);                // сигналим что можно отправлять следующий пакет и мы готовы ловить ответ
 
             // 'xQueueReceive' will "sleep" untill an event messages arrives from the RX event queue
-            if(xQueueReceive(rx_msg_q, reinterpret_cast<void*>(&event), (portTickType)portMAX_DELAY)) {
+            if(xQueueReceive(rx_msg_q, reinterpret_cast<void*>(&event), portMAX_DELAY)) {
 
                 //Handle received event
                 switch(event.type) {
@@ -426,7 +426,7 @@ private:
         // Task runs inside Infinite loop
         for (;;){
             // 'xQueueReceive' will "sleep" untill some message arrives from the msg queue
-            if(xQueueReceive(tx_msg_q, &(msg), (portTickType)portMAX_DELAY)) {
+            if(xQueueReceive(tx_msg_q, &(msg), portMAX_DELAY)) {
 
                 // if smg would expect a reply than I need to grab a semaphore from the RX queue task
                 if (msg->w4rx){

--- a/src/pzem_edl.cpp
+++ b/src/pzem_edl.cpp
@@ -187,15 +187,6 @@ void PZ003::resetEnergyCounter(){
 
 /*   === PZPool immplementation ===   */
 
-/**
- * @brief Destroy the PZPool::PZPool object
- * All registered devices and ports are destructed
- */
-PZPool::~PZPool(){
-    meters.clear();
-    ports.clear();
-}
-
 bool PZPool::addPort(uint8_t _id, UART_cfg &portcfg, const char *descr){
     if (port_by_id(_id))
         return false;       // port with such id already exist
@@ -208,10 +199,8 @@ bool PZPool::addPort(std::shared_ptr<PZPort> port){
     if (port_by_id(port->id))
         return false;       // port with such id already exist
 
-    if (!ports.add(port))
-        return false;       // some LinkedList error
-
     uint8_t portid = port->id;
+    ports.emplace_back(port);
 
     // RX handler lambda catches port-id here and suppies this id to the handler function
     port->q->attach_RX_hndlr([this, portid](RX_msg *msg){
@@ -282,15 +271,16 @@ bool PZPool::addPZEM(const uint8_t port_id, PZEM *pz){
 
     node->pzem.reset(std::move(pz));
 
-    return meters.add(node);
+    meters.emplace_back(std::move(node));
+    return true;
 }
 
 bool PZPool::removePZEM(const uint8_t pzem_id){
-    for (int i = 0; i != meters.size(); ++i) {
-        if (meters[i]->pzem->id == pzem_id){
-            meters.unlink(i);
+    for (auto i = meters.begin(); i != meters.end(); ++i ){
+        if ((*i)->pzem->id == pzem_id){
+            meters.erase(i);
             return true;
-        }
+        }        
     }
     return false;
 }

--- a/src/pzem_edl.hpp
+++ b/src/pzem_edl.hpp
@@ -15,7 +15,7 @@ GitHub: https://github.com/vortigont/pzem-edl
 #include "freertos/FreeRTOS.h"
 #include "freertos/timers.h"
 #include "pzem_modbus.hpp"
-#include "LList.h"
+#include <list>
 
 #define POLLER_PERIOD       PZEM_REFRESH_PERIOD         // auto update period in ms
 #define POLLER_MIN_PERIOD   2*PZEM_UART_TIMEOUT         // minimal poller period
@@ -340,15 +340,14 @@ class PZPool {
     };
 
 protected:
-    LList<std::shared_ptr<PZPort>> ports;                           // list of registered ports
-    LList<std::shared_ptr<PZNode>> meters;                          // list of registered PZEM nodes
+    std::list< std::shared_ptr<PZPort> > ports;                           // list of registered ports
+    std::list< std::shared_ptr<PZNode> > meters;                          // list of registered PZEM nodes
     std::shared_ptr<PZPort> port_by_id(uint8_t id);
     const PZEM* pzem_by_id(uint8_t id) const;
 
 
 public:
-    PZPool(){}
-    ~PZPool();
+    PZPool() = default;
     // Copy semantics : not implemented
     PZPool(const PZPool&) = delete;
     PZPool& operator=(const PZPool&) = delete;

--- a/src/timeseries.hpp
+++ b/src/timeseries.hpp
@@ -30,7 +30,7 @@ GitHub: https://github.com/vortigont/pzem-edl
 #include <esp_heap_caps.h>
 
 #if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
-#include "esp_psram.h"
+#include <esp_psram.h>
 #elif ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 0, 0)
 #include <esp32/spiram.h>
 #else
@@ -214,7 +214,7 @@ public:
 
 // Unary predicate for ID match
 template <class T>
-class MatchID : public std::unary_function<T, bool>{
+class MatchID {
     uint8_t _id;
 public:
     explicit MatchID(uint8_t id) : _id(id) {}
@@ -284,7 +284,7 @@ template <typename T>
 class TSContainer {
 
 public:
-    TSContainer<T>(){};
+    //TSContainer<T>(){};
     //~TSContainer();
 
     /**

